continuous_queries.md

```diff
@@ -0,0 +1,236 @@
+
+By selecting specific fields, `INTO` can merge many series into one that will go into a new either a new measurement, retention policy, or database. For example:
+
+```sql
+SELECT mean(value) as value, region
+INTO 1h.cpu_load
+FROM cpu_load
+GROUP BY time(1h), region
+```
+
+That will give 1h summaries of the mean value of the `cpu_load` for each `region`. Specifying `region` in the `GROUP BY` clause is unnecessary since having it in the `SELECT` clause forces it to be grouped by that tag, we've just included it in the example for clarity.
+
+With `SELECT ... INTO`, fields will be written as fields and tags will be written as tags.
+
+### Continuous Query Syntax
+
+The `INTO` queries run once. Continuous queries will turn `INTO` queries into something that run in the background in the cluster. They're kind of like triggers in SQL.
+
+```sql
+CREATE CONTINUOUS QUERY 1h_cpu_load
+ON database_name
+BEGIN
+  SELECT mean(value) as value, region
+  INTO 1h.cpu_load
+  FROM cpu_load
+  GROUP BY time(1h), region
+END
+```
+
+Or chain them together:
+
+```sql
+CREATE CONTINUOUS QUERY 10m_event_count
+ON database_name
+BEGIN
+  SELECT count(value)
+  INTO 10m.events
+  FROM events
+  GROUP BY time(10m)
+END
+
+-- this selects from the output of one continuous query and outputs to another series
+CREATE CONTINUOUS QUERY 1h_event_count
+ON database_name
+BEGIN
+  SELECT sum(count) as count
+  INTO 1h.events
+  FROM events
+  GROUP BY time(1h)
+END
+```
+
+Or multiple aggregations from all series in a measurement. This example assumes you have a retention policy named `1h`.
+
+```sql
+CREATE CONTINUOUS QUERY 1h_cpu_load
+ON database_name
+BEGIN
+  SELECT mean(value), percentile(80, value) as percentile_80, percentile(95, value) as percentile_95
+  INTO 1h.cpu_load
+  FROM cpu_load
+  GROUP BY time(1h), *
+END
+```
+
+The `GROUP BY *` indicates that we want to group by the tagset of the points written in. The same tags will be written to the output series. The multiple aggregates in the `SELECT` clause (percentile, mean) will be written in as fields to the resulting series.
+
+Showing what continuous queries we have:
+
+```sql
+LIST CONTINUOUS QUERIES
+```
+
+Dropping continuous queries:
+
+```sql
+DROP CONTINUOUS QUERY <name>
+ON <database>
+```
+
+### Security
+
+To create or drop a continuous query, the user must be an admin.
+
+### Limitations
+
+In order to prevent cycles and endless copying of data, the following limitation is enforced on continuous queries at create time:
+
+*The output of a continuous query must go to either a different measurement or to a different retention policy.*
+
+In theory they'd still be able to create a cycle with multiple continuous queries. We should check for these and disallow.
+
+## Proposed Architecture
+
+Continuous queries should be stored in the metastore cluster wide. That is, they amount to database schema that should be stored in every server in a cluster.
```

Comment:
I guess I thought of "schema" as specifically (just) the structure of the data within the database.